tools/migration/ppaas-artifact-converter/src/main/java/org/wso2/ppaas/tools/artifactmigration/Transformation.java

```
@@ -0,0 +1,399 @@
+            synchronized (Transformation.class) {
+                if (instance == null) {
+                    instance = new Transformation();
+                }
+            }
+        }
+
+        return instance;
+    }
+
+    /**
+     * Method to transform Auto Scale Policies
+     */
+    public void transformAutoscalePolicyList() {
+
+        List<AutoscalePolicy> autoscalePolicy400List;
+        AutoscalePolicyBean autoscalePolicy410Template;
+
+        try {
+
+            //Retrieving Json files of PPaaS 4.0.0
+            autoscalePolicy400List = OldArtifactLoader.getInstance().fetchAutoscalePolicyList();
+            log.info("Fetched Auto Scale Policy from PPaaS 4.0.0");
+
+            //Retrieving the template
+            autoscalePolicy410Template = TemplateLoader.getInstance()
+                    .fetchTemplate(Constants.ROOT_TEMPLATE_DIRECTORY + Constants.DIRECTORY_TEMPLATE_POLICY_AUTOSCALE,
+                            AutoscalePolicyBean.class);
+            log.info("Fetched Auto Scale Policy Template");
+
+            //Modifying the template according to the values of PPaaS 4.0.0
+            File directoryName = new File(Constants.ROOT_DIRECTORY + Constants.DIRECTORY_POLICY_AUTOSCALE);
+
+            for (AutoscalePolicy autoscalePolicy : autoscalePolicy400List) {
+
+                autoscalePolicy410Template.setId(autoscalePolicy.getId());
+
+                String json = new GsonBuilder().setPrettyPrinting().create().toJson(autoscalePolicy410Template);
+                String fileName = autoscalePolicy410Template.getId();
+
+                writeFile(directoryName, fileName, json);
+            }
+            log.info("Created Auto Scale Policy 4.1.0 artifacts");
+
+        } catch (JsonSyntaxException e) {
+            log.error("JSON syntax exception in fetching auto scale policies", e);
+        } catch (IOException e) {
+            log.error("IO exception in fetching auto scale policies", e);
+        }
```

I think we need to re-throw these exceptions, otherwise it will not be propagated to the caller.